udata/core/dataset/permissions.py

```diff
@@ -7,31 +7,31 @@
     OrganizationAdminNeed, OrganizationEditorNeed
 )
 
-from .models import CommunityResource
+from .models import Resource
 
 
-class DatasetEditPermission(Permission):
-    def __init__(self, dataset):
+class OwnablePermission(Permission):
+    '''A generic permission for ownable objects (with owner or organization)'''
+    def __init__(self, obj):
         needs = []
 
-        if dataset.organization:
-            needs.append(OrganizationAdminNeed(dataset.organization.id))
-            needs.append(OrganizationEditorNeed(dataset.organization.id))
-        elif dataset.owner:
-            needs.append(UserNeed(dataset.owner.id))
-
-        super(DatasetEditPermission, self).__init__(*needs)
-
-
-class ResourceEditPermission(Permission):
-    def __init__(self, resource, dataset):
-        needs = []
-        obj = resource if isinstance(resource, CommunityResource) else dataset
-
         if obj.organization:
             needs.append(OrganizationAdminNeed(obj.organization.id))
             needs.append(OrganizationEditorNeed(obj.organization.id))
         elif obj.owner:
             needs.append(UserNeed(obj.owner.id))
 
-        super(ResourceEditPermission, self).__init__(*needs)
+        super(OwnablePermission, self).__init__(*needs)
+
+
+class DatasetEditPermission(OwnablePermission):
+    '''Permissions to edit a Dataset'''
+    pass
+
+
+class ResourceEditPermission(OwnablePermission):
+    '''Permissions to edit a Resource (aka. its dataset) or community resource'''
```

Comment:
No, the point is that this permission is currently used for both community resource and resource vie its dataset (explaining the docstring and the exception I added to prevent misuse until then)